packages/cadl-ranch-specs/http/type/model/inheritance/single-discriminator/main.tsp

```diff
@@ -0,0 +1,171 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+
+using TypeSpec.Http;
+
+@doc("Illustrates inheritance with single discriminator.")
+@scenarioService("/type/model/inheritance/single-discriminator")
+namespace Type.Model.Inheritance.SingleDiscriminator;
+
+@doc("This is base model for polymorphic single level inheritance with a discriminator.")
+@discriminator("kind")
+model Bird {
+  kind: string;
+  wingspan: int32;
+}
+
+@doc("The second level model in polymorphic single level inheritance.")
+model SeaGull extends Bird {
+  kind: "seagull";
+}
+
+@doc("The second level model in polymorphic single level inheritance.")
+model Sparrow extends Bird {
+  kind: "sparrow";
+}
+
+@doc("The second level model in polymorphic single level inheritance.")
+model Goose extends Bird {
+  kind: "goose";
+}
+
+@doc("The second level model in polymorphic single levels inheritance which contains references to other polymorphic instances.")
+model Eagle extends Bird {
+  kind: "eagle";
+  friends?: Bird[];
+  hate?: Record<Bird>;
+  partner?: Bird;
+}
+
+@doc("Define a base class in the legacy way. Discriminator property is not explicitly defined in the model.")
+@discriminator("kind")
+model Dinosaur {
+  size: int32;
+}
```

Comment:
What do you mean by "the legacy way"?

TypeSpec doc https://microsoft.github.io/typespec/standard-library/built-in-decorators#examples-1
```ts
@discriminator("kind")
model Pet{ kind: string }

model Cat extends Pet {kind: "cat", meow: boolean}
model Dog extends Pet  {kind: "dog", bark: boolean}
```

Swagger 2.0
https://swagger.io/specification/v2/
> Adds support for polymorphism. The discriminator is the schema property name that is used to differentiate between other schema that inherit this schema. The property name used MUST be defined at this schema and it MUST be in the required property list. When used, the value MUST be the name of this schema or any schema that inherits it.

So I think TypeSpec at least "recommend" to have discriminator defined as property, and Swagger requires so.

Reply:
I think thats the point of this test, it is a valid representation without the `kind:` property on the base class. It might not be recommended but it is still valid.

Reply:
I see. Just wondering why "the legacy way" come from.

Tried on playground, it will produce valid Swagger. So, Yeah, I am OK with it in test case.

Reply:
it used to be the only way before we supported overriding properties in model inheritance

```
model Foo {
a: string}

model Bar extends Foo {
a: string; // this used to be an error, you couldn't redefine properties from parent models
}
```